### ISSUE TITLE
Neural network architecture image in Week 6 Policy Based is missing

### ISSUE THREAD

**Issue by nambrot** [yandexdataschool/Practical_RL#70]
As the title says, the image is missing. I've tried various architectures now but have not been able to make it work

https://github.com/yandexdataschool/Practical_RL/blob/master/week6_policy_based/homework_tensorflow.ipynb

**Comment by justheuristic** [comment]
It should work fine with the architecture you used for DQN, except for the final layers (single linear dense for q-values should be changed to policy and value layer__s__ for adv. actor-critic)

Alternatively, try [this](https://github.com/yandexdataschool/Practical_RL/blob/master/week7_pomdp/img1.jpg) except with no recurrent memory

**Comment by nambrot** [comment]
Thanks for the always prompt response!

**Comment by BoYanSTKO** [comment]
@nambrot 
based on my understanding, actor and critic have the same network body, it's just in the output layer, we output both actions and state values. so maybe just one network and the densely connected layer would be:
```python
# logits for n_actions and 1 unit for the state value
self.model.add(Dense(n_actions+1, activation='linear'))
```
Just my understanding (might be wrong). Love to hear other comments.

**Comment by nambrot** [comment]
Did that work for you? I also tried a architecture with mostly shared network, but it didn't converge either

**Comment by BoYanSTKO** [comment]
I can reach a mean reward of 400+ using a shared network. It would be great if someone could confirm if a shared network is the way to implement it.

**Comment by nambrot** [comment]
Do you mind sharing your network or giving some hints? Did you basically use the same architecture as from the DQN excercise?

**Comment by BoYanSTKO** [comment]
yes, I used the same network as the DQN exercise one, except I have both actions and states in the output layers.

**Comment by nambrot** [comment]
It worked! Though I have to admit, i'm not sure why

**Comment by BoYanSTKO** [comment]
how did you do it? Did you use a shared network like what I did or something else?
I know some people used:
```python
out1 = Dense(n_actions)(x)
out2 = Dense(1)(x)
self.network = Model(inp, [out1, out2])
```
instead of:
```python
self.network.add(Dense(n_actions+1, activation='linear'))
```

**Comment by justheuristic** [comment]
(The image has been added recently, sorry for such a long reaction. The discussion in here also seems to have concluded)
